### PR TITLE
fix(`server/aws-lambda`): Update cookie splitting for lambda adapter

### LINKED ISSUE #6456: bug: Setting cookies with expiry in lambda are broken
### Provide environment information

  System:
    OS: macOS 14.7.2
    CPU: (12) arm64 Apple M3 Pro
    Memory: 231.94 MB / 36.00 GB
    Shell: 5.9 - /bin/zsh
  Binaries:
    Node: 22.12.0 - ~/.proto/shims/node
    Yarn: 4.1.0 - ~/.proto/shims/yarn
    npm: 10.4.0 - ~/.proto/shims/npm
    pnpm: Checking for package overrides
9.15.0 - ~/.proto/shims/pnpm
    bun: 1.0.32 - ~/.bun/bin/bun
  Browsers:
    Chrome: 133.0.6943.53
    Safari: 18.2
  npmPackages:
    typescript: ^5.7.3 => 5.7.3 

### Describe the bug

When setting a cookie which has an expiry eg `Sun, 15 Jul 2012 00:00:01 GMT` the lambda adapter breaks the cookie into 2 resulting in the cookie not being set with any params after the expiry including the expiry 

### To reproduce

Apply a set cookie with an expiry date and deploy to lambda, the reproduction needs to be deployed to lambda. 

### Additional information

The issue was introduced here https://github.com/trpc/trpc/pull/5783/files#diff-a1587748fc98224d3b2ef8f04104b72b5b029f913587da98272dc53fa3df3d5bR48 and is still in next https://github.com/trpc/trpc/blob/caed91ee782f78ada6c8c0c82e2d9d2ff474f555/packages/server/src/adapters/aws-lambda/getPlanner.ts#L48

It's unsafe to just split the cookies on comma without considering the expiry field. The set cookie header should be parsed and then split into the individual cookies

### 👨‍👧‍👦 Contributing

- [ ] 🙋‍♂️ Yes, I'd be down to file a PR fixing this bug!

### PR DESCRIPTION
**Pull request by cmanou**
Closes #6456

## 🎯 Changes

Updated to use cookie-es to correctly handle the splitting safely for values with expiry. Also the initial test cases weren't actually valid cookies as they by minimum needed the = sign so updated them. 

## ✅ Checklist

- [x] I have followed the steps listed in the [Contributing guide](https://github.com/trpc/trpc/blob/main/CONTRIBUTING.md).
- [x] If necessary, I have added documentation related to the changes made.
- [x] I have added or updated the tests related to the changes made.

## Summary by CodeRabbit

- **New Features**
  - Enhanced cookie handling now accurately extracts detailed attributes from HTTP cookie headers—including values, expiration dates, and domain settings—to improve session management.
  
- **Bug Fixes**
  - Addressed issues with parsing cookie header strings, ensuring cookies with complex formatting are consistently and correctly processed.

### CODE REVIEW

File: packages/server/package.json [hunk 1]
```diff
@@ -126,6 +126,7 @@
     "@types/react": "^19.0.0",
     "@types/react-dom": "^19.0.0",
     "@types/ws": "^8.2.0",
+    "cookie-es": "^1.2.2",
```
Comment (KATT) [inline packages/server/package.json:129]:
```suggestion
```

**Review by KATT — CHANGES_REQUESTED** [review]
Please do this without a library or vendoring the part that is needed from the lib.

This won't work unless you install cookie-es in your project

**Comment by cmanou** [comment]
@KATT updated, apologies I had a brief look at the rollup config and was thinking it was only marking certain packages as externals  not all listed packages. Have just ventured the split and added a comment for the licenses on top

File: packages/server/src/adapters/aws-lambda/getPlanner.ts [hunk 2]
```diff
@@ -43,10 +44,9 @@ interface Processor<TEvent extends LambdaEvent> {
 function getHeadersAndCookiesFromResponse(response: Response) {
   const headers = Object.fromEntries(response.headers.entries());
 
-  const cookies: string[] = response.headers
-    .getSetCookie()
-    .flatMap((value) => value.split(','))
-    .map((cookie) => cookie.trim());
+  const cookies: string[] = splitSetCookieString(
+    response.headers.getSetCookie(),
+  ).map((cookie) => cookie.trim());
```
Comment (cmanou) [inline packages/server/src/adapters/aws-lambda/getPlanner.ts:49]: I don't think the trim is actually necessary my guess is it used to be for space separation with the comma splitting  but leaving it for now just in case someone was relying on it.

**Review by KATT — APPROVED** [review]
